pkg/environment/environment.go

```diff
@@ -127,10 +137,12 @@ func GetEnv() EnvVariables {
 
 // ValidateEnv validates environment variables.
 func ValidateEnv(env EnvVariables) error {
-	if env.EnableDeployAppGateway && len(env.AppGwSubnetID) == 0 {
-		return errors.New("Missing required Environment variables: Provide APPGW_SUBNETID (ENVT001)")
-	} else if len(env.SubscriptionID) == 0 || len(env.ResourceGroupName) == 0 || len(env.AppGwName) == 0 {
-		return errors.New("Missing required Environment variables: Provide APPGW_SUBSCRIPTION_ID, APPGW_RESOURCE_GROUP and APPGW_NAME (ENVT002)")
+	if len(env.AppGwName) == 0 {
+		return errors.New("Missing required Environment variables: Provide atleast provide APPGW_NAME. You can also provided APPGW_SUBSCRIPTION_ID and APPGW_RESOURCE_GROUP (ENVT001)")
+	}
+	if env.EnableDeployAppGateway && (len(env.AppGwResourceID) == 0 && len(env.AppGwSubnetPrefix) == 0 && len(env.AppGwSubnetName) == 0) {
+		// when create is true, then either we should have env.AppGwSubnetID or env.AppGwSubnetPrefix
+		return errors.New("Missing required Environment variables: Please provide APPGW_SUBNETNAME or APPGW_SUBNETID of an existing subnet. If you want AGIC to optionally create a new subnet, then also provide APPGW_SUBNETPREFIX (ENVT002)")
```

Comment:
At some point it might be good to break this into 3 separate error statements.
Also it'd be good to pull the `errors.New()` out into errors.go